pkg/wasm/imagefetcher.go

```diff
@@ -135,14 +135,14 @@ func (o *ImageFetcher) Fetch(url, expManifestDigest string) ([]byte, error) {
 	)
 }
 
-func parseReference(url string) (name.Reference, error) {
+func (o *ImageFetcher) parseReference(url string) (name.Reference, error) {
 	ref, err := name.ParseReference(url)
 	if err != nil {
 		return nil, err
 	}
 
 	// fallback to http based request, inspired by [helm](https://github.com/helm/helm/blob/12f1bc0acdeb675a8c50a78462ed3917fb7b2e37/pkg/registry/client.go#L594)
-	_, err = remote.Get(ref)
+	_, err = remote.Get(ref, o.fetchOpts...)
```

Comment:
At line 151, should we return nil instead err, if preflight get check result in some other error? i.e. in this function we will only deal with https fallback instead of attributing all other type of errors to URL parsing error.

Reply:
seems reasonable